File: ZemljaSlova.Services/Database/Event.cs
using System;
using System.Collections.Generic;

namespace ZemljaSlova.Services.Database;

public partial class Event
{
    public int Id { get; set; }

    public string Title { get; set; } = null!;

    public string Description { get; set; } = null!;

    public string? Location { get; set; }

    public DateTime StartAt { get; set; }

    public DateTime EndAt { get; set; }

    public string? Organizer { get; set; }

    public string? Lecturers { get; set; }

    public byte[]? CoverImage { get; set; }

    public int? MaxNumberOfPeople { get; set; }

    public virtual ICollection<TicketType> TicketTypes { get; set; } = new List<TicketType>();
}

File: UI/zemljaslova_desktop/lib/models/event.dart
import 'package:json_annotation/json_annotation.dart';
import 'ticket_type.dart';

part 'event.g.dart';

@JsonSerializable()
class Event {
  final int id;
  final String title;
  final String description;
  final String? location;
  final DateTime startAt;
  final DateTime endAt;
  final String? organizer;
  final String? lecturers;
  final String? coverImageUrl;
  final int? maxNumberOfPeople;
  final List<TicketType>? ticketTypes;

  Event({
    required this.id,
    required this.title,
    required this.description,
    this.location,
    required this.startAt,
    required this.endAt,
    this.organizer,
    this.lecturers,
    this.coverImageUrl,
    this.maxNumberOfPeople,
    this.ticketTypes,
  });

  factory Event.fromJson(Map<String, dynamic> json) => _$EventFromJson(json);

  Map<String, dynamic> toJson() => _$EventToJson(this);
} 